clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1482,7 +1486,7 @@ private void updatePatternSubscription(Cluster cluster) {
                 .collect(Collectors.toSet());
         if (subscriptions.subscribeFromPattern(topicsToSubscribe)) {
             applicationEventHandler.add(new SubscriptionChangeEvent());
-            metadata.requestUpdateForNewTopics();
+            this.metadataVersionSnapshot = metadata.requestUpdateForNewTopics();
```

Comment:
Since we're completing the regex implementation, could you clean up the func doc? We should remove the TODO, and I would suggest we rephrase the description, and simply state what this evaluates the regex against the list of topic names from metadata, and updates the subscription state with the matching topics.